packages/smooth_app/lib/pages/all_product_list_page.dart

```diff
@@ -62,39 +61,42 @@ class AllProductListModal extends StatelessWidget {
                     subtitle: Text(
                       appLocalizations.user_list_length(productsLength),
                     ),
-                    trailing: PopupMenuButton<PopupMenuEntries>(
+                    trailing: PopupMenuButton<ProductListPopupMenuEntry>(
                       itemBuilder: (BuildContext context) {
-                        return <PopupMenuEntry<PopupMenuEntries>>[
-                          _shareMenu(
-                            appLocalizations,
-                            daoProductList,
-                            localDatabase,
-                            context,
-                            productList,
-                          ),
-                          _openInWebMenu(
-                            appLocalizations,
-                            daoProductList,
-                            localDatabase,
-                            context,
-                            productList,
-                          ),
-                          if (productsLength > 0)
-                            _clearListMenu(
-                              appLocalizations,
-                              daoProductList,
-                              localDatabase,
-                              context,
-                              productList,
-                            ),
-                          if (productList.isEditable)
-                            _deleteListMenu(
-                              appLocalizations,
-                              daoProductList,
-                              context,
-                              productList,
-                            ),
+                        final bool enableRename =
+                            productList.listType == ProductListType.USER;
+                        final List<ProductListPopupItem> list =
+                            <ProductListPopupItem>[
+                          if (enableRename) ProductListPopupRename(),
+                          ProductListPopupShare(),
+                          ProductListPopupOpenInWeb(),
+                          if (productsLength > 0) ProductListPopupClear(),
+                          if (productList.isEditable) ProductListPopupDelete(),
                         ];
+                        final List<PopupMenuEntry<ProductListPopupMenuEntry>>
```

Comment:
Why not use the `map` operator here instead?

Reply:
I tried to find something relevant but couldn't find anything.
I'm not at ease with those map/list methods anyway.
My code can be read by anyone, uses OOP and has no copy/paste, which is good enough for me.